### PR TITLE
[fix]: [PL-28610]: Added GitX thread decorator from Request body for new Pipeline APIs

### PR DESCRIPTION
**Pull request by mankrit-singh**
## Describe your changes
Added GitX thread decorator from Request body for new Pipeline APIs

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/38315)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/38315)

**Comment by mankrit-singh** [comment]
trigger messagecheck

File: pipeline-service/service/src/main/java/io/harness/pms/pipeline/api/GitXThreadDecorator.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+import io.harness.context.GlobalContext;
+import io.harness.gitsync.beans.StoreType;
+import io.harness.gitsync.interceptor.GitEntityInfo;
+import io.harness.gitsync.interceptor.GitSyncBranchContext;
+import io.harness.manage.GlobalContextManager;
+import io.harness.spec.server.pipeline.model.GitCreateDetails;
+import io.harness.spec.server.pipeline.model.GitUpdateDetails;
+
+public class GitXThreadDecorator {
```
Comment (namanvermah) [inline pipeline-service/service/src/main/java/io/harness/pms/pipeline/api/GitXThreadDecorator.java:18]: We can move these methods to GitAwareContextHelper actually, no need to create a new class

Comment (mankrit-singh) [reply pipeline-service/service/src/main/java/io/harness/pms/pipeline/api/GitXThreadDecorator.java:18]: sure

**Comment by mohitg0795** [comment]
*[`878-ng-common-utilities/src/main/java/io/harness/gitaware/helper/GitAwareContextHelper.java` line 130 at r2](https://reviewable.io/reviews/harness/harness-core/38315#-NDXYeNZ5eqGOiasnDh6:-NDXYeNZ5eqGOiasnDh7:b-iap3q3) ([raw file](https://github.com/harness/harness-core/blob/0db9fa0866d044084a1fde2559e883bb10a3552d/878-ng-common-utilities/src/main/java/io/harness/gitaware/helper/GitAwareContextHelper.java#L130)):*
> ```Java
>     }
>     GlobalContextManager.upsertGlobalContextRecord(GitSyncBranchContext.builder().gitBranchInfo(gitEntityInfo).build());
>   }
> ```

line space missing

**Comment by mohitg0795** [comment]
*[`878-ng-common-utilities/src/main/java/io/harness/gitaware/helper/GitAwareContextHelper.java` line 131 at r2](https://reviewable.io/reviews/harness/harness-core/38315#-NDXYurw9gfHn7J6q_8i:-NDXYurw9gfHn7J6q_8j:bbbasvf) ([raw file](https://github.com/harness/harness-core/blob/0db9fa0866d044084a1fde2559e883bb10a3552d/878-ng-common-utilities/src/main/java/io/harness/gitaware/helper/GitAwareContextHelper.java#L131)):*
> ```Java
>     GlobalContextManager.upsertGlobalContextRecord(GitSyncBranchContext.builder().gitBranchInfo(gitEntityInfo).build());
>   }
>   public static void populateGitDetailsUpdate(GitUpdateDetails gitDetails) {
> ```

can't we have a single method with a common arg which everyone uses? is there some specific use-case that we need to solve for create and update flows?

**Comment by mankrit-singh** [comment]
*[`878-ng-common-utilities/src/main/java/io/harness/gitaware/helper/GitAwareContextHelper.java` line 131 at r2](https://reviewable.io/reviews/harness/harness-core/38315#-NDXYurw9gfHn7J6q_8i:-NDXZcL5BL3VSo0Q6PS3:bif92lw) ([raw file](https://github.com/harness/harness-core/blob/0db9fa0866d044084a1fde2559e883bb10a3552d/878-ng-common-utilities/src/main/java/io/harness/gitaware/helper/GitAwareContextHelper.java#L131)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

can't we have a single method with a common arg which everyone uses? is there some specific use-case that we need to solve for create and update flows?
</blockquote></details>

Making a related change

**Review by archit-harness — APPROVED** [review]
> Reviewed 1 of 3 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38315)* status: 4 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by namanvermah — APPROVED** [review]
> Reviewed 1 of 3 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38315)* status: 1. unresolved discussions.

**Review by namanvermah — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38315#-:-NDajdAT2LfyY_6nVHwm:bnfp4nl)* status: 1. unresolved discussions.

**Comment by mohitg0795** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/api/PipelinesApiImpl.java` line 103 at r4](https://reviewable.io/reviews/harness/harness-core/38315#-NDbj6dG3sxmpEj32zdI:-NDbj6dG3sxmpEj32zdJ:b-iph88m) ([raw file](https://github.com/harness/harness-core/blob/7797f6967068f7c003dda18cde382571e3bffcc6/pipeline-service/service/src/main/java/io/harness/pms/pipeline/api/PipelinesApiImpl.java#L103)):*
> ```Java
>   @Override
>   @NGAccessControlCheck(resourceType = "PIPELINE", permission = PipelineRbacPermissions.PIPELINE_VIEW)
>   public Response getPipeline(@OrgIdentifier String org, @ProjectIdentifier String project,
> ```

@mankrit-singh just curious, why is the ordering of identifiers like this? Is it intentional or a mistake that its org->project->acc ?

Ideally, I believe we should follow acc->org->project->any other id

**Comment by mohitg0795** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/api/PipelinesApiImpl.java` line 185 at r4](https://reviewable.io/reviews/harness/harness-core/38315#-NDbjL724LNOxhQIZjr6:-NDbjL733uXTRdDASGlQ:b-4iz9di) ([raw file](https://github.com/harness/harness-core/blob/7797f6967068f7c003dda18cde382571e3bffcc6/pipeline-service/service/src/main/java/io/harness/pms/pipeline/api/PipelinesApiImpl.java#L185)):*
> ```Java
>   @NGAccessControlCheck(resourceType = "PIPELINE", permission = PipelineRbacPermissions.PIPELINE_CREATE_AND_EDIT)
>   public Response updatePipeline(PipelineUpdateRequestBody requestBody, @OrgIdentifier String org,
>       @ProjectIdentifier String project, @ResourceIdentifier String pipeline, @AccountIdentifier String account) {
> ```

here as well, the ordering looks too random. @archit-harness @mankrit-singh is this as per expectations?

**Comment by mankrit-singh** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/api/PipelinesApiImpl.java` line 185 at r4](https://reviewable.io/reviews/harness/harness-core/38315#-NDbjL724LNOxhQIZjr6:-NDbjVTN0CITj3iLnfcN:b-5hbbln) ([raw file](https://github.com/harness/harness-core/blob/7797f6967068f7c003dda18cde382571e3bffcc6/pipeline-service/service/src/main/java/io/harness/pms/pipeline/api/PipelinesApiImpl.java#L185)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

here as well, the ordering looks too random. @archit-harness @mankrit-singh is this as per expectations?
</blockquote></details>

Yes, this is because of code generation from spec, cant change it as of now

**Comment by mankrit-singh** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/api/PipelinesApiImpl.java` line 103 at r4](https://reviewable.io/reviews/harness/harness-core/38315#-NDbj6dG3sxmpEj32zdI:-NDbjfG38oKk7X6MIGOo:b-o38hfi) ([raw file](https://github.com/harness/harness-core/blob/7797f6967068f7c003dda18cde382571e3bffcc6/pipeline-service/service/src/main/java/io/harness/pms/pipeline/api/PipelinesApiImpl.java#L103)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

@mankrit-singh just curious, why is the ordering of identifiers like this? Is it intentional or a mistake that its org->project->acc ?

Ideally, I believe we should follow acc->org->project->any other id
</blockquote></details>

it is because of code generation

**Comment by mohitg0795** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/api/PipelinesApiUtils.java` line 368 at r4](https://reviewable.io/reviews/harness/harness-core/38315#-NDbkjOj0Vf6vJlloFOi:-NDbkjOk6Oc-tuJh6Orw:b-22bpyh) ([raw file](https://github.com/harness/harness-core/blob/7797f6967068f7c003dda18cde382571e3bffcc6/pipeline-service/service/src/main/java/io/harness/pms/pipeline/api/PipelinesApiUtils.java#L368)):*
> ```Java
>         .repoName(gitDetails.getRepoName())
>         .build();
>   }
> ```

spacing issue

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 1 of 3 files at r2, 1 of 3 files at r3, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38315)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Review by archit-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38315)* status: :shipit: complete! ✓

**Pull request merged**